### ISSUE TITLE
TypeError: global_add_pool() missing 1 required positional argument: 'batch'

### ISSUE THREAD

**Issue by Rajjaa** [pyg-team/pytorch_geometric#4809]
### 🐛 Describe the bug

The documentation of `global_add_pool ` states that the batch parameter is optional, however I get this error ```TypeError: global_add_pool() missing 1 required positional argument: 'batch'``` when I don't pass a value for the batch. 

Here is a snippet of code to reproduce the bug:
```Python
import torch
from torch_geometric.nn import global_add_pool
x = torch.zeros(4,5)
z = global_add_pool(x)
```

I think the origin of the issue is the absence of a default value for the batch parameter in the definition of `global_add_pool`.
And the same bug happens when using `global_mean_pool` or `global_max_pool`.

### Environment

* PyG version: 2.0.5
* PyTorch version: 1.11.0
* OS:  Fedora release 35
* Python version: 3.9
* CUDA/cuDNN version: 11.3
* How you installed PyTorch and PyG (`conda`, `pip`, source): Poetry

**Comment by rusty1s** [comment]
Yes, this is intended. I feel that `global_add_pool(x)` can lead to some serious bugs when not being careful, as such:
```python
def forward(x, edge_index, batch=None):
    x = global_ad__pool(x, batch)
```
is a good and more robust workaround. WDYT?

**Comment by RexYing** [comment]
I think both suggestions are great. We could adjust the documentation to specify that batch is mendatory? And similarly for other global pooling functions too.

The ability to handle both sparse and dense batches sounds great too. As you said this seems to be easy with torch sum. Should we have a PR on this?

**Comment by rusty1s** [comment]
We should be able to do this by doing a dense aggregation in `dim=-2` instead of `dim=0`. Happy to accept a PR on this one.

**Comment by rusty1s** [comment]
Fixed in #4827.
